tests/components/modem_callerid/test_init.py

```diff
@@ -28,28 +32,26 @@ async def test_async_setup_entry_not_ready(hass: HomeAssistant):
     """Test that it throws ConfigEntryNotReady when exception occurs during setup."""
     entry = MockConfigEntry(
         domain=DOMAIN,
-        data=CONF_DATA,
+        data={CONF_DEVICE: com_port().device},
     )
     entry.add_to_hass(hass)
 
-    with patch(
-        "homeassistant.components.modem_callerid.PhoneModem",
-        side_effect=exceptions.SerialError(),
-    ):
+    with patch_init_modem() as modemmock:
+        modemmock.side_effect = exceptions.SerialError
         await hass.config_entries.async_setup(entry.entry_id)
     assert len(hass.config_entries.async_entries(DOMAIN)) == 1
-    assert entry.state == ConfigEntryState.SETUP_ERROR
+    assert entry.state == ConfigEntryState.SETUP_RETRY
```

Comment:
If this is only a refactor, why has the result changed here?

Reply:
The patch target has changed. Only `ConfigEntryNotReady` is raised in `__init__.py` now.